dali/test/python/test_operator_random_object_bbox.py

```diff
@@ -191,16 +191,29 @@ def generate_data(shape, num_classes, blobs_per_class):
 
     return label
 
+def generate_samples(num_samples, ndim, dtype):
+    samples = []
+    for i in range(num_samples):
+        shape = list(np.random.randint(5, 13, [ndim]))
+        num_classes = np.random.randint(1, 10)
+        blobs_per_class = np.random.randint(1, 10);
+        samples.append(generate_data(shape, num_classes, blobs_per_class).astype(dtype))
+    return samples
+
+
 def batch_generator(batch_size, ndim, dtype):
+    """Returns a generator that generates completely new data each time it's called"""
     def gen():
         # batch_size = np.random.randint(1, max_batch_size+1)
-        batch = []
-        for i in range(batch_size):
-            shape = list(np.random.randint(5, 13, [ndim]))
-            num_classes = np.random.randint(1, 10)
-            blobs_per_class = np.random.randint(1, 10);
-            batch.append(generate_data(shape, num_classes, blobs_per_class).astype(dtype))
-        return batch
+        return generate_samples(batch_size, ndim, dtype)
+    return gen
+
+def sampled_dataset(dataset_size, batch_size, ndim, dtype):
+    """Returns a generator that returns random samples from a pre-generated dataset"""
+    data = generate_samples(dataset_size, ndim, dtype)
+    def gen():
+        # batch_size = np.random.randint(1, max_batch_size+1)
```

Comment:
remove?

Reply:
I keep it so we can restore it when batch size propagation from batched to per-sample external source finally works.